resource/resource.go

```diff
@@ -39,8 +39,14 @@ func (res Resource) Validate() error {
 	}
 	// TODO(ericsnow) Ensure Revision is 0 for OriginUpload?
 
-	if err := res.Fingerprint.Validate(); err != nil {
-		return errors.Annotate(err, "bad fingerprint")
+	if res.Fingerprint.IsZero() {
+		if res.Size > 0 {
+			return errors.NewNotValid(nil, "missing fingerprint; sized resources must have one")
```

Comment:
error text is kind of awkward.  0 is a size ;)  

I think it's sufficient to say "missing fingerprint".